### PR TITLE
[ES-48] Add multikey signer content

### PR DESCRIPTION
**Pull request by bgravenorst**
Signed-off-by: Byron Gravenorst byron.gravenorst@consensys.net

## Pull Request Description
Add multikey signer content.

## Fixed Issue(s)
ES-48

### CODE REVIEW

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -287,11 +287,11 @@ Name of the vault to access. Sub-domain of `vault.azure.net`.
 --keyvault-name="MyKeyVault"
 ```
 
-## Multifile Options
+## Multikey Options
 
 ### directory
 
-Path to the directory containing the key and password files.
+Path to the directory containing the [TOML files which define parameters](../Multifile-Parameters.md) required to access key files.
```
Comment (macfarla) [inline docs/Reference/CLI/CLI-Syntax.md]:
```suggestion
Path to the directory containing the [TOML files which define parameters](../Multifile-Parameters.md) required to access key credentials.
```

File: docs/Reference/Multifile-Parameters.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
+description: TOML file parameters for multi signers.
+---
+
+# Multikey Parameters
+
+Create a TOML configuration file for each key that signs transactions. Keys can be stored in a V3 keystore file, Hashicorp Vault, and Azure Key Vault.
```
Comment (macfarla) [inline docs/Reference/Multifile-Parameters.md]:
```suggestion
Create a TOML configuration file for each key that signs transactions. Keys can be stored in a V3 keystore file, Hashicorp Vault, or Azure Key Vault.
```

File: docs/Reference/Multifile-Parameters.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    auth-file = "/Users/me/project/hashicorp_auth"
+    timeout = 500
+    ```
+
+!!! note
+    We recommend using an absolute path when specifying `auth-file`. Relative paths are located relative to where EthSigner is executed.
+  
+| Key                  | Description                           |
+|----------------------|---------------------------------------|
+| **type**             | Type of key signing. Use `hashicorp-based-signer`|
```
Comment (macfarla) [inline docs/Reference/Multifile-Parameters.md]:
```suggestion
| **type**             | Type of key signing. Use `hashicorp-signer`|  
```

File: docs/Reference/Multifile-Parameters.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    key-vault-name = "AzureKeyVault"
+    key-name = "ethsignerKey"
+    key-version = "7c01fe58d68148bba5824ce418241092"
+    client-id = "47efee5c-8079-4b48-31bb4f2e9a22"
+    client-secret = "TW_3Uc/HLPdpLp5*om@MGcd1T29MuP*5"
+    ```
+
+| Key                  | Description                           |
+|----------------------|---------------------------------------|
+| **type**             | Type of key signing. Use `azure-based-signer`|
```
Comment (macfarla) [inline docs/Reference/Multifile-Parameters.md]:
```suggestion
| **type**             | Type of key signing. Use `azure-signer`|  
```

File: docs/Reference/Multifile-Parameters.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    client-id = "47efee5c-8079-4b48-31bb4f2e9a22"
+    client-secret = "TW_3Uc/HLPdpLp5*om@MGcd1T29MuP*5"
+    ```
+
+| Key                  | Description                           |
+|----------------------|---------------------------------------|
+| **type**             | Type of key signing. Use `azure-based-signer`|  
+| **key-vault-name**   | Name of the vault to access. Sub-domain of `vault.azure.net` |
+| **key-name**         | Name of key to be used |
+| **key-version**      | Version of the specified key to use |
```
Comment (macfarla) [inline docs/Reference/Multifile-Parameters.md]:
```suggestion
| **key-version**      | Version of the specified key |
```

File: mkdocs.yml [hunk 6]
```diff
@@ -72,6 +72,7 @@ nav:
         - Store Multiple Key Files: Concepts/Multiple-Key-Files.md
     - Reference:
         - EthSigner Command Line: Reference/CLI/CLI-Syntax.md
+        - Multikey TOML Parameters: Reference/Multifile-Parameters.md
```
Comment (macfarla) [inline mkdocs.yml]: should this be renamed to Reference/Multikey-Parameters.md

File: docs/Concepts/Multiple-Key-Files.md [hunk 7]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
 
 # Store Multiple Key Files
 
-EthSigner supports file-based signing using multiple V3 keystore files. Each account that signs
-transactions requires a key file.
+Each account that signs transactions requires a key file. EthSigner supports [multiple key files](../Tutorials/Multifile.md) that can be stored in:
```
Comment (macfarla) [inline docs/Concepts/Multiple-Key-Files.md]:
```suggestion
Each account that signs transactions requires a key file. EthSigner supports [multiple keys](../Tutorials/Multifile.md) that can be stored in:
```

Comment (macfarla) [inline docs/Concepts/Multiple-Key-Files.md]:
```suggestion
Each account that signs transactions requires a key. EthSigner supports [multiple keys](../Tutorials/Multifile.md) that can be stored in:
```

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (macfarla) [reply docs/Reference/CLI/CLI-Syntax.md]: or maybe just "required to access keys"

File: docs/Concepts/Multiple-Key-Files.md [hunk 8]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
...
-The key file name must contain the account address and follow the `[<prefix>]<accountAddress>.key`
-format. The password file must have an identical name and end with the `.password` extension.
-No two key files can have the same key address in the file name, even if the prefix differs.
+Each key requires a separate [TOML file](../Reference/Multifile-Parameters.md) which defines the parameters to access the key. The TOML files must
+be placed in a single directory specified using the [`multikey-signer`](../Reference/CLI/CLI-Syntax.md#multikey-options) subcommand.
 
-In the example the account address is prefixed with the date. The `0x` portion of the account address
-must be removed.
+!!! note
+    New files can be dropped into the directory without needing to restart EthSigner.
```
Comment (macfarla) [inline docs/Concepts/Multiple-Key-Files.md]: Similarly, files can be removed without needing to restart EthSigner

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Syntax.md]: Done.

File: docs/Reference/Multifile-Parameters.md (hunk 2 above)
Comment (bgravenorst) [reply docs/Reference/Multifile-Parameters.md]: Done

File: docs/Reference/Multifile-Parameters.md (hunk 3 above)
Comment (bgravenorst) [reply docs/Reference/Multifile-Parameters.md]: done

File: docs/Reference/Multifile-Parameters.md (hunk 4 above)
Comment (bgravenorst) [reply docs/Reference/Multifile-Parameters.md]: done

File: docs/Reference/Multifile-Parameters.md (hunk 5 above)
Comment (bgravenorst) [reply docs/Reference/Multifile-Parameters.md]: Done

File: docs/Concepts/Multiple-Key-Files.md (hunk 7 above)
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

File: mkdocs.yml (hunk 6 above)
Comment (bgravenorst) [reply mkdocs.yml]: Done

File: docs/Reference/Multikey-Parameters.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
+---
+description: TOML file parameters for multi signers.
+---
+
+# Multikey Parameters
+
+Create a TOML configuration file for each key that signs transactions. Keys can be stored in a V3 keystore file, Hashicorp Vault, or Azure Key Vault.
```
Comment (MadelineMurray) [inline docs/Reference/Multikey-Parameters.md]: This paragraph can go into the new Using Multiple Signing Keys How To topic - it's not really reference content

File: docs/Reference/Multikey-Parameters.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+The filename must be formatted as `[<prefix>]<accountAddress>.toml` and must be located in the directory
```
Comment (MadelineMurray) [inline docs/Reference/Multikey-Parameters.md]: Same comment as previous paragraph

File: docs/Reference/Multikey-Parameters.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+---
+
+# Multikey Parameters
+
+Create a TOML configuration file for each key that signs transactions. Keys can be stored in a V3 keystore file, Hashicorp Vault, or Azure Key Vault.
+
+The filename must be formatted as `[<prefix>]<accountAddress>.toml` and must be located in the directory 
+specified using the [`multikey-signer --directory`](../CLI/CLI-Syntax#multikey-options) command line option.
+
+The optional `[metadata]` section in the TOML files is used for data that is not required by EthSigner.
```
Comment (MadelineMurray) [inline docs/Reference/Multikey-Parameters.md]: _is used for_ -> _contains_ 

Suggested edit to reduce number of words

File: docs/Tutorials/Multifile.md [hunk 12]
```diff
@@ -6,6 +6,8 @@ description: Signing transactions with multiple keys.
 
 EthSigner supports file-based signing using [multiple keys](../Concepts/Multiple-Key-Files.md).
 
+This tutorial covers configuring multiple keys using a V3 keystore file. To configure multiple keys for Hashicorp Vault or Azure Key Vault, update the [TOML configuration file](../Reference/Multikey-Parameters.md) accordingly.
```
Comment (MadelineMurray) [inline docs/Tutorials/Multifile.md]: Suggest adding links to the Store Key Externally -> Use Hashicorp Vault and Use Azure Key Vault topics on Hashicorp Vault and Azure Key Vault

File: docs/Concepts/Overview.md [hunk 13]
```diff
@@ -15,8 +15,7 @@ EthSigner can sign transactions with a key stored in:
 * [Azure Key Vault](../HowTo/Store-Keys/Use-Azure.md).
 
 !!! note
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Overview.md]: Done.

File: docs/Concepts/Multiple-Key-Files.md [hunk 14]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
 
 # Store Multiple Key Files
 
-EthSigner supports file-based signing using multiple V3 keystore files. Each account that signs
-transactions requires a key file.
+Each account that signs transactions requires a key. EthSigner supports [multiple keys](../Tutorials/Multifile.md) that can be stored in:
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

File: docs/Concepts/Multiple-Key-Files.md [hunk 15]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
...
 
-The key file must follow the required naming convention, and have an accompanying password file.
-[Create the required files in a directory](../Tutorials/Multifile.md) and use the
-[multifile-based-signer](../Reference/CLI/CLI-Syntax.md#multifile-options) subcommand to specify
-the directory location.
+* A V3 keystore file stored on a file system accessible by the host.
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

File: docs/Concepts/Multiple-Key-Files.md [hunk 16]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
...
+* [Hashicorp Vault](../HowTo/Store-Keys/Use-Hashicorp.md).
+* [Azure Key Vault](../HowTo/Store-Keys/Use-Azure.md).
 
-!!! note
-    EthSigner does not support using multiple key signers with Hashicorp Vault or Azure Key Vault.
-
-The key file name must contain the account address and follow the `[<prefix>]<accountAddress>.key`
-format. The password file must have an identical name and end with the `.password` extension.
-No two key files can have the same key address in the file name, even if the prefix differs.
+Each key requires a separate [TOML file](../Reference/Multikey-Parameters.md) which defines the parameters to access the key. The TOML files must
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

File: docs/Concepts/Multiple-Key-Files.md [hunk 17]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
...
+be placed in a single directory specified using the [`multikey-signer`](../Reference/CLI/CLI-Syntax.md#multikey-options) subcommand.
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

File: docs/Concepts/Multiple-Key-Files.md [hunk 18]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
...
 
-In the example the account address is prefixed with the date. The `0x` portion of the account address
-must be removed.
+!!! note
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done

File: docs/Concepts/Multiple-Key-Files.md [hunk 19]
```diff
@@ -4,28 +4,23 @@ description: Signing transactions with multiple keys.
...
+    Files can be added or removed from the directory without needing to restart EthSigner.
 
-!!! example
-    `10-10-2019_78e6e236592597c09d5c137c2af40aecd42d12a2.key`
+The TOML file name must use the format `[<prefix>]<accountAddress>.toml`.
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Concepts/Multiple-Key-Files.md]: Done.

File: docs/Reference/Multikey-Parameters.md (hunk 9 above)
Comment (bgravenorst) [reply docs/Reference/Multikey-Parameters.md]: Done

File: docs/Reference/Multikey-Parameters.md (hunk 10 above)
Comment (bgravenorst) [reply docs/Reference/Multikey-Parameters.md]: Done

File: docs/Reference/Multikey-Parameters.md (hunk 11 above)
Comment (bgravenorst) [reply docs/Reference/Multikey-Parameters.md]: Done

File: docs/Tutorials/Multifile.md (hunk 12 above)
Comment (bgravenorst) [reply docs/Tutorials/Multifile.md]: Done

**Pull request merged**